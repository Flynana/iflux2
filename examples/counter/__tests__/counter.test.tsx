import * as React from 'react'
import * as renderer from 'react-test-renderer'
import Counter from '../src/component/counter'
import Store from '../src/store'
import { StoreProvider } from 'iflux2'

@StoreProvider(Store)
class App extends Counter {
  render() {
    return super.render();
  }
}

describe('Relax Counter', () => {
  it('test counter', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // tree.props.increment();
    window['store'].increment();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
