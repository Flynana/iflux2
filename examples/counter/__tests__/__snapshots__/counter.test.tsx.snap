exports[`Relax Counter test counter 1`] = `
<div>
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    increment
  </a>
  <br />
  <span>
    0
  </span>
  <br />
  <span>
    QL:0
  </span>
  <br />
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    decrement
  </a>
</div>
`;

exports[`Relax Counter test counter 2`] = `
<div>
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    increment
  </a>
  <br />
  <span>
    1
  </span>
  <br />
  <span>
    QL:1
  </span>
  <br />
  <a
    href="javascript:void(0);"
    onClick={[Function]}>
    decrement
  </a>
</div>
`;
